crates/algod_client/src/apis/simulate_transaction.rs

```diff
@@ -0,0 +1,105 @@
+/*
+ * Algod REST API.
+ *
+ * API endpoint for algod operations.
+ *
+ * The version of the OpenAPI document: 0.0.1
+ * Contact: contact@algorand.com
+ * Generated by: Rust OpenAPI Generator
+ */
+
+use algokit_http_client::HttpClient;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use super::{AlgodApiError, ContentType, Error};
+use algokit_transact::AlgorandMsgpack;
+
+// Import all custom types used by this endpoint
+use crate::models::{ErrorResponse, SimulateTransaction};
+
+// Import request body type if needed
+use crate::models::SimulateRequest;
+
+/// struct for typed errors of method [`simulate_transaction`]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum SimulateTransactionError {
+    Status400(ErrorResponse),
+    Status401(ErrorResponse),
+    Status500(ErrorResponse),
+    Status503(ErrorResponse),
+    Statusdefault(),
+    DefaultResponse(),
+    UnknownValue(serde_json::Value),
+}
+
+/// Simulates a raw transaction or transaction group as it would be evaluated on the network. The simulation will use blockchain state from the latest committed round.
+pub async fn simulate_transaction(
+    http_client: &dyn HttpClient,
+    request: SimulateRequest,
+    format: Option<&str>,
```

Comment:
Enum rather than string?

Reply:
@neilcampbell added generalized handler to templates that will create typed enums for any params from oas spec where input is a string with explicit set of acceptable literals